apps/ledger-live-mobile/src/screens/Protect/RedirectToRecoverStaxFlow.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useCallback, useEffect } from "react";
+import { DeviceModelId } from "@ledgerhq/types-devices";
+import { Flex } from "@ledgerhq/native-ui";
+import { useFeature } from "@ledgerhq/live-common/featureFlags/index";
+import { Linking } from "react-native";
+import { BaseNavigatorStackParamList } from "../../components/RootNavigator/types/BaseNavigator";
+import {
+  RootComposite,
+  StackNavigatorProps,
+} from "../../components/RootNavigator/types/helpers";
+import { ScreenName } from "../../const";
+import { useNavigationInterceptor } from "../Onboarding/onboardingContext";
+import BleDevicePairingFlow from "../../components/BleDevicePairingFlow";
+import DeviceSetupView from "../../components/DeviceSetupView";
+import { useIncrementOnNavigationFocusState } from "../../helpers/useIncrementOnNavigationFocusState";
+import { usePostOnboardingURI } from "../../hooks/recoverFeatureFlag";
+import { ServicesConfig } from "../../components/ServicesWidget/types";
+
+type NavigationProps = RootComposite<
+  StackNavigatorProps<
+    BaseNavigatorStackParamList,
+    ScreenName.RedirectToRecoverStaxFlow
+  >
+>;
+
+export function RedirectToRecoverStaxFlowScreen({
+  navigation,
+}: NavigationProps) {
+  const { setShowWelcome, setFirstTimeOnboarding } = useNavigationInterceptor();
+  const recoverRestoreFlowURI = usePostOnboardingURI();
+  const recoverConfig: ServicesConfig | null = useFeature(
+    "protectServicesMobile",
+  );
+
+  useEffect(() => {
+    setShowWelcome(false);
+    setFirstTimeOnboarding(false);
+  }, [setFirstTimeOnboarding, setShowWelcome]);
+
+  // Makes sure the pairing components are reset when navigating back to this screen
+  const keyToReset =
+    useIncrementOnNavigationFocusState<NavigationProps["navigation"]>(
+      navigation,
+    );
+
+  const onPairingSuccess = useCallback(() => {
+    if (recoverConfig?.enabled && recoverRestoreFlowURI) {
+      Linking.canOpenURL(recoverRestoreFlowURI).then(() =>
+        Linking.openURL(recoverRestoreFlowURI),
+      );
+    }
+  }, [recoverConfig?.enabled, recoverRestoreFlowURI]);
+
+  const filterByDeviceModelId = DeviceModelId.nanoX;
+  const areKnownDevicesDisplayed = true;
```

Comment:
hm maybe just set the value of the props directly in the JSX instead of declaring variables here no?